library/x509_crt.c

```diff
@@ -3005,6 +3005,127 @@ static int x509_crt_verify_chain(
     }
 }
 
+#ifdef _WIN32
+/* ??? */
+#elif defined(__sun)
+/* Solaris requires -lsocket -lnsl for inet_pton() */
+#elif defined(__has_include)
+#if __has_include(<sys/socket.h>)
+#include <sys/socket.h>
+#endif
+#if __has_include(<arpa/inet.h>)
+#include <arpa/inet.h>
+#endif
+#else
+#include <sys/socket.h>
+#include <arpa/inet.h>
```

Comment:
I wrote above:
>  I have pushed an additional change to attempt to include `<sys/socket.h>` and `<arpa/inet.h>` if `__has_include()` is not present.  Let's see if the CI like it on the test platforms.

It sounds like you would prefer my prior commits, which used my local implementation if `__has_include()` was not supported by the compiler.  I had changed that because you were concerned about relying on a compiler feature for platform support.

I'll be adding some test for the local implementation and will change the code back to use the local implementation if the compiler does not support `__has_include()`

Reply:
CI passes except for `all_u16-build_armcc`:
`Error:  #5: cannot open source input file "sys/socket.h": No such file or directory`
I could add an exception to use the local inet_pton implementation when `defined(__ARMCC_VERSION)`

Instead, I have done the following, as I think you are requesting:
```
--- a/library/x509_crt.c
+++ b/library/x509_crt.c
@@ -3016,9 +3016,6 @@ find_parent:
 #if __has_include(<arpa/inet.h>)
 #include <arpa/inet.h>
 #endif
-#else
-#include <sys/socket.h>
-#include <arpa/inet.h>
 #endif

 #ifndef AF_INET6
```
:rocket: